### ISSUE TITLE
Download errors

### ISSUE THREAD

**Issue by Destro9000** [vipulyaara/Kafka#19]
Hello developer, i have some problems when trying to download any file on kafka , it just keeps failing after 1 min or so , could you solve this problem ?

**Comment by Destro9000** [comment]
Ok so , the download keeps going a bit and stops and everytime i have to reclick the download button.
Another issue is that leaving the app stops it from downloading and this is annoying , if you could fix this thanks

**Comment by vipulyaara** [comment]
Thank you for filing the issue. Download speed is the only issue blocking us from reaching 1.0.0. The files are hosted on not-so-fast servers and that combined with the lack of advanced download frameworks in Android makes it a really hard problem. I tried different solutions but there was no significant improvement. I am still looking into it.

**Comment by vipulyaara** [comment]
One solution I am investigating now is [FDM](https://www.freedownloadmanager.org/download-fdm-for-android.htm). It improves download speed multi-folds but doesn't provide an SDK. A quick solution can be to ask the user to install FDM and then deep link into their app but I don't think users would like that.

**Comment by Destro9000** [comment]
@vipulyaara i understand , having to install another app just to download won't be appreciated ,especially by people with low memory in their phone

**Comment by Destro9000** [comment]
If i have like 1DM does it work with the downloads? It should be a download manager

**Comment by vipulyaara** [comment]
The app doesn't support opening files downloaded externally so it would not show up in Kafka. We can implement that if we decide to support external downloaders but I am not yet sure that's the best way.

**Comment by Destro9000** [comment]
Ok thank you for the info

**Comment by dschwen** [comment]
It would already help, if the app pressed the download button again for me if the download was interrupted and didn't reach 100%...

**Comment by vipulyaara** [comment]
> It would already help, if the app pressed the download button again for me if the download was interrupted and didn't reach 100%...

Yes, retrying automatically (e.g. 3 times) is a good idea. I'll implement this.

**Comment by vipulyaara** [comment]
This is implemented. The downloader now retries failed downloads 3 times. It should be available in the next release. Download timeout is also extended from 3 minutes to 7 minutes.
